### PR TITLE
Adding AccTauDEM

### PR DESCRIPTION
**Pull request by RyanMcCarthy-NOAA**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/acctaudem/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+build:
+  number: 0
+  skip: true  # [win or osx]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - mpich
```
Comment (scopatz) [inline recipes/acctaudem/meta.yaml:21]: mpich is probably not needed in build, only in host

File: recipes/acctaudem/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - make
+    - cmake
+  host:
+    - libgdal
+    - mpich
```
Comment (scopatz) [inline recipes/acctaudem/meta.yaml:26]: Is this mpich-only?  You should look at https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi

Comment (RyanMcCarthy-NOAA) [reply recipes/acctaudem/meta.yaml:25]: The README says to use mpich. It also says openmpi is an option, but that is it untested. So I figured it say to stick with mpich.

**Comment by RyanMcCarthy-NOAA** [comment]
@scopatz did my comments on `mpich` make sense for this PR?

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**